generator/Generator.sc

```diff
@@ -2925,157 +2930,168 @@ def generateTestClasses(): Unit = {
+                    final $SeqType<String> empty = $StreamType.empty();
+                    try {
+                        Match(empty).of(
+                                Case($$($ListType.empty()), ignored -> "list")
+                        );
+                        fail("expected MatchError");
+                    } catch (MatchError err) {
+                        // ok!
+                    }
+                }
 
-              xs"""
                 @$test
-                public void shouldMatchPattern$i() {
-                    final Tuple$i<${(1 to i).gen(j => s"Integer")(", ")}> tuple = Tuple.of(${(1 to i).gen(j => s"1")(", ")});
-                    final String func = Match(tuple).of(
-                            Case($$Tuple$i($d(0)${(2 to i).gen(j => s", $d()")}), (${(1 to i).gen(j => s"m$j")(", ")}) -> "fail"),
-                            Case($$Tuple$i(${(1 to i).gen(j => s"$d()")(", ")}), (${(1 to i).gen(j => s"m$j")(", ")}) -> "okFunc")
-                    );
-                    assertThat(func).isEqualTo("okFunc");
-                    final String supp = Match(tuple).of(
-                            Case($$Tuple$i($d(0)${(2 to i).gen(j => s", $d()")}), () -> "fail"),
-                            Case($$Tuple$i(${(1 to i).gen(j => s"$d()")(", ")}), () -> "okSupp")
-                    );
-                    assertThat(supp).isEqualTo("okSupp");
-                    final String val = Match(tuple).of(
-                            Case($$Tuple$i($d(0)${(2 to i).gen(j => s", $d()")}), "fail"),
-                            Case($$Tuple$i(${(1 to i).gen(j => s"$d()")(", ")}), "okVal")
-                    );
-                    assertThat(val).isEqualTo("okVal");
-
-                    final ClzMatch c = new ClzMatch2();
-                    final String match = Match(c).of(
-                            Case(Match.Pattern$i.of(ClzMatch1.class, ${(1 to i).gen(j => s"$d()")(", ")}, t -> Tuple.of(${(1 to i).gen(j => s"null")(", ")})), "fail"),
-                            Case(Match.Pattern$i.of(ClzMatch2.class, ${(1 to i).gen(j => s"$d()")(", ")}, t -> Tuple.of(${(1 to i).gen(j => s"null")(", ")})), "okMatch")
-                    );
-                    assertThat(match).isEqualTo("okMatch");
+                public void shouldCatchClassCastExceptionWhenPredicateHasDifferentType() {
+                    try {
+                        final Object o = "";
+                        Match(o).of(
+                                Case($$((Integer i) -> true), "never")
+                        );
+                        fail("expected MatchError");
+                    } catch (MatchError err) {
+                        // ok!
+                    }
                 }
-              """
-            })("\n\n")}
+
+            }
+
+            @Nested
+            class MatchPatterns {
+
+                class ClzMatch {}
+                class ClzMatch1 extends ClzMatch {}
+                class ClzMatch2 extends ClzMatch {}
```

Comment:
I changed these to non-static since static inner classes are not supported in JDK 8